src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -986,24 +986,32 @@ type internal TypeCheckInfo
                 None
 
     /// Suggest name based on type
-    let SuggestNameBasedOnType g pos ty =
-        if isNumericType g ty then
-            CreateCompletionItemForSuggestedPatternName pos "num"
-        else
-            match tryTcrefOfAppTy g ty with
-            | ValueSome tcref when not (tyconRefEq g g.system_Object_tcref tcref) ->
-                CreateCompletionItemForSuggestedPatternName pos tcref.DisplayName
-            | _ -> None
+    let SuggestNameBasedOnType (g: TcGlobals) pos ty =
+        match ty with
```

Comment:
Should match on type directly, should either use `isAppTy`, `tryAppTy` or at the very least `stripTyEqns`.

Reply:
If I do that, the abbreviation is stripped away.